bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_generic.py

```diff
@@ -44,3 +44,8 @@ def pythonizegeneric(klass, name):
     AddPrettyPrintingPyz(klass)
 
     return True
+
+import cppyy
+cppyy.gbl.DeclareGenericCallable = GetCppCallableClass("Generic")
```

Comment:
What is the difference between `Generic` and `Proxy`?

Reply:
Discussed with @stwunsch face to face.
Agreed solution:
- One decorator: `@CppCallable` with one optional parameter `numba_only`, with default false
- By default, the behaviour would be to try numba first and, if that fails, silently continue and try the slow option.
- If the user specifies `numba_only`, numba will be tried and if it fails, an error will be thrown.
- In the cpp world, the wrapper of the python callable will be called `CppCallable::f` so that users can easily relate it to its Python counterpart:
```python
@CppCallable(...)
def f():
  ...
```

Reply:
Just realized that we cannot have the decorator and the namespace be called `CppCallable` because cppyy will create the namespace proxy with the same name than the free function of the callable (which obviously crashes).

So I've decided to call the operator `@DeclareCppCallable`, which is reasonably close.

Reply:
I'll introduce a further argument `generic_only`, to force the slow implementation. Mainly because of debugging and testing reasons.

Reply:
> Just realized that we cannot have the decorator and the namespace be called `CppCallable` because cppyy will create the namespace proxy with the same name than the free function of the callable (which obviously crashes).
> 
> So I've decided to call the operator `@DeclareCppCallable`, which is reasonably close.

Not if the decorator is in `ROOT.callables` or similar instead of in the global namespace, as suggested in a previous comment. If we can avoid long names like `@DeclareCppCallable` I believe it would be better.

Reply:
> I'll introduce a further argument `generic_only`, to force the slow implementation. Mainly because of debugging and testing reasons.

Ok good, we need to check if both `generic_only` and `numba_only` are set to true and complain in that case.